### PR TITLE
[Login] Bad login experience if slow network connection

### PR DESCRIPTION
**Pull request by scottliubox**

### CODE REVIEW

**Comment by boxcla** [comment]
Hi @scottliubox, thanks for the pull request. Before we can merge it, we need you to sign our Contributor License Agreement. You can do so electronically here: http://opensource.box.com/cla

Once you have signed, just add a comment to this pull request saying, "CLA signed". Thanks!

**Comment by scottliubox** [comment]
CLA signed

**Comment by boxcla** [comment]
Verified that @scottliubox has just signed the CLA. Thanks, and we look forward to your contribution.

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m [hunk 1]
```diff
@@ -238,6 +264,8 @@ - (BOOL)webView:(UIWebView *)webView shouldStartLoadWithRequest:(NSURLRequest *)
         return NO;
     }
 
+    [self.activityIndicator startAnimating];
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m:257]: Do we need it here if it's in didstartload? (or vice versa)

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m [hunk 2]
```diff
@@ -45,6 +45,9 @@ @interface BOXAuthorizationViewController () <UIWebViewDelegate, NSURLConnection
 @property (nonatomic, readwrite, assign) NSInteger ntlmAuthFailures;
 @property (nonatomic, readwrite, assign) NSInteger authChallengeCycles;
 
+@property (nonatomic, readwrite, assign) BOOL hasFinishedLoadingLoginPage;
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m]: Do we need this?

File: BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m [hunk 3]
```diff
@@ -118,6 +123,15 @@ - (void)loadView
 	self.view = webView;
 }
 
+- (void)viewDidAppear:(BOOL)animated
```
Comment (boxtcarpel) [inline BoxContentSDK/BoxContentSDK/OAuth2/BOXAuthorizationViewController.m]: could we remove this method and have the same behavior?

**Comment by boxtcarpel** [comment]
+2 LGTM

:mobile_phone_off:

**Comment by scottliubox** [comment]
Ignore this PR, refer to:
https://github.com/box/box-ios-sdk/pull/96

**Pull request closed without merging**